example/vrops_mqtt.py

```diff
@@ -0,0 +1,122 @@
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF       #
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS   #
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN    #
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)    #
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF     #
+#  THE POSSIBILITY OF SUCH DAMAGE.                                            #
+# ----------------------------------------------------------------------------#
+
+from liota.transports.mqtt import Mqtt
+from liota.boards.gateway_dk300 import Dk300
+from liota.dcc.vrops import Vrops
+from liota.transports.web_socket import WebSocket
+from liota.things.device import Device
+import Queue
+
+# getting values from conf file
+config = {}
+execfile('sampleProp.conf', config)
+# queue defined to store the values
+# from the subscribed MQTT Channel
+glass_panel_queue = Queue.Queue()
+solar_lights_queue = Queue.Queue()
+
+# Add message callbacks that will trigger on a specific subscription match.
+# Values collected from the callback function are stored in the respective queue
+# Callback function for glass panel
+def callback_glass_panel(client, userdata, message):
+    glass_panel_queue.put(float(message.payload))
+
+# Callback function for solar lights
+def callback_solar_lights(client, userdata, message):
+    solar_lights_queue.put(float(message.payload))
+
+#---------------------------------------------------------------------------
+# In this example, we demonstrate how an event stream of data from MQTT channel
+# can be directed to vrops data center component using Liota by setting
+# sampling_interval_sec parameter to zero. Additionally, we are showcasing that
+#  with help of lambda function how we can pass parameters to commmon user defined method
+# Common User Defined Method (UDM) returning the value from respective queue
+def get_value(queue):
+    return queue.get(block=True)
+
+if __name__ == '__main__':
+    # create a mqtt_conn object by passing the port number and the server address
+    # we will be enhancing this in future to support username/password as well as QoS option
+    # currently we support QoS=1
+    mqtt_conn = Mqtt(config['MqttServer'], config['MqttPort'])
+    # subscribe to the base mqtt channel example "SmartBuilding/#"
+    mqtt_conn.subscribe(config['MqttChannel'])
+    # add a message callback to the client of mqtt_conn object
+    # glass panel call back method added to the mqtt client to listen
+    # on specific sub-channel example "SmartBuilding/GlassPanel"
+    mqtt_conn.client.message_callback_add(config['MqttSubChannel1'], callback_glass_panel)
+    # solar lights call back method added to the mqtt client
+    # to listen on specific sub-channel example "SmartBuilding/SolarLights"
+    mqtt_conn.client.message_callback_add(config['MqttSubChannel2'], callback_solar_lights)
```

Comment:
Ln 77 and Ln 80 operate on mqtt_conn.client directly from the user app, which does not look elegant.